src/TraceEvent/TraceEvent.cs

```diff
@@ -3103,7 +3103,17 @@ public static TraceEventDispatcher GetDispatcherFromFileName(string traceFileNam
             }
             else if (traceFileName.EndsWith(".btl", StringComparison.OrdinalIgnoreCase))
             {
-                return new BPerfEventSource(traceFileName);
+                if (options.StartTime == default(DateTime))
+                {
+                    options.StartTime = DateTime.MinValue;
+                }
+
+                if (options.EndTime == default(DateTime))
+                {
+                    options.EndTime = DateTime.MaxValue;
+                }
```

Comment:
Should this be inside of the BPerfEventSource ctor so that if the event source is used directly this check is executed?